axiom/optimizer/ToGraph.cpp

```diff
@@ -1247,6 +1247,29 @@ void extractNonInnerJoinEqualities(
 
 } // namespace
 
+void ToGraph::addJoinColumns(
+    const logical_plan::LogicalPlanNode& joinSide,
+    ColumnVector& columns,
+    ExprVector& exprs) {
+  const auto channels = usedChannels(joinSide);
+  const auto& names = joinSide.outputType()->names();
+  for (auto i = 0; i < channels.size(); ++i) {
+    const auto& name = names[channels[i]];
```

Comment:
Please use range based for when possible

```suggestion
  const auto& names = joinSide.outputType()->names();
  for (auto channel : usedChannels(joinSide)) {
    const auto& name = names[channel];
```